Packages/MovieDBData/Sources/MovieDBData/DataSources/Remote/API/Movies/MoviesRemoteDataSource.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  MoviesRemoteDataSource.swift
+//  MovieDB
+//
+//  Created by MOUAD BENJRINIJA on 14/3/2023.
+//
+
+import Foundation
+import Combine
+import MovieDBCore
+
+// MARK: Protocol
+public protocol MoviesRemoteDataSource: RemoteDataSource {
+  func fetchMovies(page: Int) async throws -> PaginatedList<Movie>
+  func fetchMovieDetails(id: Int) async throws -> MovieDetails
+}
+
+// MARK: Main Impl
+public struct MoviesRemoteDataSourceMain: MoviesRemoteDataSource {
+  public let session: URLSession
+
+  public init(session: URLSession) {
+    self.session = session
+  }
+  
+  public func fetchMovies(page: Int) async throws -> PaginatedList<Movie> {
+    try await fetch(call: Call.getMovies(page: page))
+       .map { (response: MoviesListRemoteModel) -> PaginatedList<Movie> in
+         response.transform()
+      }
+      .eraseToAnyPublisher()
+      .async()
+  }
+
+  public func fetchMovieDetails(id: Int) async throws -> MovieDetails {
+    try await fetch(call: Call.getMovieDetails(id: id))
+      .map { (response: MovieDetailRemoteModel) -> MovieDetails in
+        response.transform()
+      }
+      .eraseToAnyPublisher()
```

Comment:
I would ask Apple about that haha, as it's still confusing what to use and where, -JK- it's actually because of the history of the project, as I said this is a refactoring of an existing project that was initially written using Combine, and then I decided to migrate to async-await so u will find residues of combine everywhere. 
However, if it's a SwiftUi app I would probably use async-await only, but I think UIKit goes well with Combine as well (imperative code support), -i might be wrong tho- so, I would like to know your preference regarding this.

Reply:
Well, Combine is more mature ofc, but it feels like Apple is moving away from it and focusing on async-await, probably because it's pretty common in almost all other languages like Javascript, C#...